### PR TITLE
Detect managed kyma

### PR DESCRIPTION
**Pull request by nesmabadr**
**Description**

Changes proposed in this pull request:

- For the deploy command, check if there is a managed Kyma deployed
- Based on this check, display a prompt for the user whether to continue with the deployment or not

**Related issue(s)**
#1472

### CODE REVIEW

File: cmd/kyma/alpha/deploy/cmd.go
Comment (Tomasz-Smelcerz-SAP) [inline cmd/kyma/alpha/deploy/cmd.go:1]: The `deploy` command is not the only one affected, the `undeploy` is another one.

File: cmd/kyma/alpha/deploy/cmd.go [hunk 1]
```diff
@@ -208,6 +209,12 @@ func (cmd *command) deploy(ctx context.Context) error {
 	}
 	clusterAccess.Successf("Successfully connected to cluster")
 
+	if !cmd.opts.CI && !cmd.opts.NonInteractive {
```
Comment (Tomasz-Smelcerz-SAP) [inline cmd/kyma/alpha/deploy/cmd.go:212]: The legacy logic for detecting managed runtime is already configured in line 129. At the time the CLI should use both approaches. In the future we'll remove the legacy `reconciler-based` approach.

Comment (nesmabadr) [reply cmd/kyma/alpha/deploy/cmd.go:212]: @Tomasz-Smelcerz-SAP this is the `Kyma alpha deploy` not the `Kyma deploy` command

Comment (Tomasz-Smelcerz-SAP) [reply cmd/kyma/alpha/deploy/cmd.go:212]: You are right. I've looked up in the `deploy` command file.
I think it's correct.

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Pull request merged**